runtime/tests/port/j9strTest.c

```diff
@@ -1528,46 +1528,56 @@ j9str_WinacpToMutf8(struct J9PortLibrary *portLibrary)
 #if defined(J9ZOS390)
 /**
  * Check whether a null/non-null terminated string is properly handled by fstring().
+ * Also checks if number, long number, long long number, and float are properly handled.
  * It calls atoe_vsnprintf() to verify the functionalities.
  * @param[in] portLibrary - The port library under test
  * @param[in] precision - flag for precision specifier
  * @param[in] buffer - pointer to the buffer intended for atoe_vsnprintf()
  * @param[in] bufferLength - the buffer length
  * @param[in] expectedOutput - The expected string
+ * @param[in] expectedCount - The expected response from atoe_vsnprintf()
  * @param[in] format - The string format
  *
  * @return TEST_PASS on success
  *         TEST_FAIL on error
  */
 static int
-printToBuffer(struct J9PortLibrary *portLibrary, BOOLEAN precision, char *buffer, size_t bufferLength, const char *expectedOutput, const char *format, ...)
+printToBuffer(struct J9PortLibrary *portLibrary, BOOLEAN precision, char *buffer, size_t bufferLength, const char *expectedOutput, int expectedCount, const char *format, ...)
 {
 	PORT_ACCESS_FROM_PORT(portLibrary);
+	static const char NULLCHARSTRING[] = "[null]";
 	int stringLength = 0;
+	int expectedLength = 0;
+	int actualCount = 0;
 	va_list args;
 
 	va_start(args, format);
-	stringLength = atoe_vsnprintf(buffer, bufferLength, format, args);
-	va_end (args);
+	actualCount = atoe_vsnprintf(buffer, bufferLength, format, args);
+	expectedLength = (NULL == expectedOutput) ? 0 : strlen(expectedOutput);
+	stringLength = (NULL == buffer) ? 0 : strlen(buffer);
+	const char *nonNullExpectedOutput = (NULL == expectedOutput) ? NULLCHARSTRING : expectedOutput;
+	const char *nonNullBuffer = (NULL == buffer) ? NULLCHARSTRING : buffer;
+	va_end(args);
 
-	if (stringLength >= 0) {
+	if (actualCount >= 0) {
 		if (precision) {
-			j9tty_printf(PORTLIB, "\n\tFinish Testing: String in buffer is: \"%s\", length = %d\n", buffer, stringLength);
+			j9tty_printf(PORTLIB, "\n\tFinish Testing: String in buffer is: \"%s\", length = %d\n", nonNullBuffer, stringLength);
 		} else {
-			j9tty_printf(PORTLIB, "\n\tFinish Testing without precision specifier: String in buffer is: \"%s\", length = %d\n", buffer, stringLength);
+			j9tty_printf(PORTLIB, "\n\tFinish Testing without precision specifier: String in buffer is: \"%s\", length = %d\n", nonNullBuffer, stringLength);
 		}
 
-		if ((stringLength == strlen(expectedOutput))
-			&& (0 == memcmp(buffer, expectedOutput, stringLength))
+		if ((stringLength == expectedLength)
+			&& (expectedCount == actualCount)
+			&& (0 == memcmp(nonNullBuffer, nonNullExpectedOutput, stringLength + 1))
```

Comment:
It is not reasonable to assume callers don't produce the output `"[null]"`; suggest:
```c
			&& ((NULL == buffer) || (0 == memcmp(buffer, expectedOutput, expectedCount + 1)))
```